js/main.js

```diff
@@ -1,45 +1,491 @@
+function playerMouseClick(evt) {
+  if (evt.target.tagName === "INPUT" || evt.target.tagName === "LABEL" || shipCounter > 6) return;
+  let clickedDiv = PLAYER_DIVS.indexOf(evt.target);
+  let rows = parseInt(clickedDiv.toString().charAt(0));
+  let columns = parseInt(clickedDiv.toString().charAt(1));
+  if (isNaN(columns)) {
+    columns = rows;
+    rows = 0;
+  }
+  let shipCanBePlaced;
+  shipCanBePlaced = canBePlaced(
+    PLAYER_BOARD,
+    rows,
+    columns,
+    verticalCheckbox === true ? "vertical" : "horizontal",
+    Object.values(playerShips)[shipCounter]
+  );
+  if (shipCanBePlaced === true) {
+    placeShips(
+      PLAYER_BOARD,
+      rows,
+      columns,
+      verticalCheckbox === true ? "vertical" : "horizontal",
+      Object.values(playerShips)[shipCounter]
+    );
+    shipCounter++;
+  }
+  activeTurn();
+}
+
+//Rendering the ships on the game board
+function renderShips(board) {
+  let shipCanBePlaced;
+  let rows;
+  let columns;
+  let idx = Math.round(Math.random());
+  const vertOrHoriz = ["vertical", "horizontal"];
+
+  Object.values(cpuShips).forEach((shipObj) => {
+    do {
+      rows = getRandomNum(0, HEIGHT);
+      columns = getRandomNum(0, LENGTH);
+      shipCanBePlaced = canBePlaced(
+        board,
+        rows,
+        columns,
+        vertOrHoriz[idx],
+        shipObj
+      );
+    } while (!shipCanBePlaced);
+    placeShips(board, rows, columns, vertOrHoriz[idx], shipObj);
+    idx = Math.round(Math.random());
+  });
+}
+
+//Modifying the header based on the current games state
+function activeTurn() {
+  if (shipCounter <= 6) {
+    activePlayerEl.innerHTML = `Place your ${
+      Object.values(playerShips)[shipCounter].name
+    } ${Object.values(playerShips)[shipCounter].length} Tiles `;
+  } else if (winner !== null) {
+    playAgainBtn.innerHTML = "Play Again";
+    activePlayerEl.innerHTML = `${winner === 1 ? "PLAYER WINS!" : "CPU WINS!"}`;
+    document.body.appendChild(playAgainBtn);
+    return;
+  } else {
+    verticalLabelEl.style.display = "none";
+    activePlayerEl.innerHTML = `${turn === 1 ? "PLAYER'S TURN" : "CPU'S turn"}`;
+  }
+  return sleep(700);
+}
+
+//Checking if the ship can be placed on the selected div
+function canBePlaced(board, rows, columns, vertOrHoriz, ship) {
+  if (vertOrHoriz === "horizontal") {
+    for (let i = columns; i < columns + ship.length; i++) {
+      if (i >= LENGTH || board[rows][i] !== 0) return false;
+    }
+  } else {
+    for (let i = rows; i < rows + ship.length; i++) {
+      if (i >= LENGTH || board[i][columns] !== 0) return false;
+    }
+  }
+  return true;
+}
+
+//Actually placing the ships after confirming that they can be placed
+function placeShips(board, rows, columns, vertOrHoriz, ship) {
+  if (vertOrHoriz === "horizontal") {
+    for (let i = columns; i < columns + ship.length; i++) {
+      board === PLAYER_BOARD
+        ? PLAYER_DIVS[rows * LENGTH + i].classList.add(
+            "player-ship",
+            "horizontal"
```

Comment:
Instead of adding `horizontal` and `vertical` classes to these divs, why not have an `orientation` property on the ships themselves that you can set?